test/Microsoft.AspNetCore.SignalR.Client.Tests/RpcConnectionFacts.cs

```diff
@@ -0,0 +1,127 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO.Pipelines;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Sockets.Client;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Xunit;
+
+namespace Microsoft.AspNetCore.SignalR.Client.Tests
+{
+    public class RpcConnectionFacts : IDisposable
+    {
+        TestServer testServer;
+        public RpcConnectionFacts()
+        {
+            var webHostBuilder = new WebHostBuilder().
+                ConfigureServices(services =>
+                {
+                    services.AddSignalR();
+                })
+                .Configure(app =>
+                {
+                    app.UseSignalR(routes =>
+                    {
+                        routes.MapHub<TestHub>("/hubs");
+                    });
+                });
+            testServer = new TestServer(webHostBuilder);
+        }
+
+        [Fact]
+        public async void CheckFixedMessage()
+        {
+            var loggerFactory = new LoggerFactory();
+
+            using (var httpClient = testServer.CreateClient())
+            using (var pipelineFactory = new PipelineFactory())
+            {
+                var transport = new LongPollingTransport(httpClient, loggerFactory);
+                using (var connection = await HubConnection.ConnectAsync(new Uri("http://test/hubs"), new JsonNetInvocationAdapter(), transport, httpClient, pipelineFactory, loggerFactory))
```

Comment:
consider creating a helper method/property for the Uri. I think you can get the first part of the uri from testServer. 

I also think that we may want to expose an overload that takes string instead of Uri (or maybe we will not even have the one that takes an uri). It's separate though.